Makefile

```diff
@@ -121,23 +128,24 @@ $(BIN_D): $(OBJS_D) | $(BINDIR)/.
 $(DEPS): ;
 include $(wildcard $(DEPS))
 
-$(OBJDIR_R)/%.o: $(SRCDIR)/%$(SRCEXT) $(DEPDIR_R)/%.d | $(OBJDIR_R)/. $(DEPDIR_R)/.
+.SECONDEXPANSION:
+$(OBJDIR_R)/%.o: $(SRCDIR)/%$(SRCEXT) $(DEPDIR_R)/%.d | $$(dir $$@). $$(dir $(DEPDIR_R)/$$*.d).
 	$(CXX) $(CXXFLAGS) $(FLAGS_R) $(INCFLAGS) -c $< -o $@ $(DEPGENFLAGS_R)
 	$(POSTCOMPILE_R)
 
-$(OBJDIR_D)/%.o: $(SRCDIR)/%$(SRCEXT) $(DEPDIR_D)/%.d | $(OBJDIR_D)/. $(DEPDIR_D)/.
+$(OBJDIR_D)/%.o: $(SRCDIR)/%$(SRCEXT) $(DEPDIR_D)/%.d | $$(dir $$@). $$(dir $(DEPDIR_D)/$$*.d).
 	$(CXX) $(CXXFLAGS) $(FLAGS_D) $(INCFLAGS) -c $< -o $@ $(DEPGENFLAGS_D)
 	$(POSTCOMPILE_D)
```

Comment:
I think there should be a newline after `.SECONDEXPANSION`, since it affects more than just the rule immediately below.
We may even want to move `make c` and `make clean` to above `.SECONDEXPANSION`, since those rules don't need it. What is your opinion on this?

Reply:
I was also considering this, I just didn't take the time to put it in a commit yet. I do agree on moving `clean` and `c`.